docs/AUTH.md

```diff
@@ -1,36 +1,387 @@
 
-- **useXXX**: Means that a resource can be used and **id**, **title**, **description** can be viewed.
-- **manageXXX**: Means that a resource can be read and modified.
-- **createXXX**: Allow the creation of a feature. 
-- **readXXX**: Allow reading of a feature.
-- **updateXXX**: Allow to update a feature.
-- **deleteXXX**: Allow to delete a feature.
-- **purgeXXX**: Allow to purge a feature, this is final deletion.
+```json
+{
+    "urm": {
+        "xyz-hub": {
+            // zero or more Actions
+            "<action>": [
+                // one or more Attribute-Map's (atleast one should match)
+                {
+                    // zero or more access Attributes (all should match)
+                    "<accessAttribute1>": "<exactValue>",
+                    "<accessAttribute2>": "<valueWithWildcard*>",
+                    "<accessAttribute3>": [  // one or more values (all should match)
+                        "<anotherExactValue>",
+                        "<anotherValueWithWildcard*>"
+                    ]
+                }
+            ]
+        }
+    }
+}
+```
 
-## Access-Matrix Types
+For example:
 
-- **EventHandler**: 
+```json
+{
+    "urm": {
+        "xyz-hub": {
+            // zero or more Actions
+            "readFeatures": [
+                // one or more Attribute-Map's (atleast one should match)
+                {
+                    // zero or more access Attributes (all should match)
+                    "id": "my-unique-feature-id",
+                    "storageId": "id-with-wild-card-*",
+                    "tags": [  // one or more values (all should match)
+                        "my-unique-tag",
+                        "some-common-tag-with-wild-card-*"
+                    ]
+                }
+            ]
+        }
+    }
+}
+```
+
+
+
+### Action
+
+Defines type of operation allowed against a resource. Refer later sections on this page, to know which specific actions are allowed against which resource.
+
+But, in general:
+
+- **useXXX**: Means that a resource can be used and **id**, **title**, **description** can be viewed, but NOT **properties** details
+- **manageXXX**: Means that a resource can be read and modified (i.e. full access).
+- **createXXX**: Allow the creation of a resource.
+- **readXXX**: Allow reading of a resource.
+- **updateXXX**: Allow to update a resource.
+- **deleteXXX**: Allow to delete a resource.
+
+Each Action can have list of one or more **AttributeMaps**, which is then compared against target Resource AttributeMaps to validate authorization.
+
+For example:
+
+```json
+{
+    "readFeatures": [
+        // one or more Attribute-Map's (atleast one should match)
+        {
+            "id": "my-unique-feature-id"
+        },
+        {
+            "storageId": "id-with-wild-card-*"
+        },
+        {
+            "tags": [
+                "my-unique-tag",
+                "some-common-tag-with-wild-card-*"
+            ]
```

Comment:
I now understand where the AND requirement comes from. From access matrix perspective each tag is an attribute. So the tags should be added into the root (because the root is AND combined). Example:
```json
{
  "tag:my-unique-tag": null,
  "tag:some-common-tag-with-wild-card-*": null
}
```
This requires some definition first, how the ARM is compared against the URM. Actually, we can define that the value _null_ in the URM means that any value is allowed. For attributes we then allow name expansion via "*", which is not hard to implement.
This clearly specifies what is AND and what is OR and keeps the whole concept compatible.

Reply:
Same comment as for "additional scope"